pytorch_lightning/trainer/trainer.py

```diff
@@ -2042,9 +2042,13 @@ def devices(self) -> Optional[Union[List[int], str, int]]:
 
     @property
     def data_parallel_device_ids(self) -> Optional[List[int]]:
-        return (
-            self._accelerator_connector.parallel_device_ids if self._accelerator_connector.parallel_device_ids else None
-        )
+        rank_zero_deprecation("`Trainer.data_parallel_device_ids` was deprecated in v1.6.")
+        if isinstance(self.accelerator, GPUAccelerator):
+            if isinstance(self.strategy, ParallelStrategy):
+                return [i for i in range(len(self.strategy.parallel_devices))]
+            elif isinstance(self.strategy, SingleDeviceStrategy):
+                return [0]
+        return None
```

Comment:
I think this should return an empty list in the base case

Reply:
The original function returns None  that's why I kept it the same. I'll change to empty list instead.